Dockerfile

```diff
@@ -0,0 +1,35 @@
+FROM jupyter/minimal-notebook
+LABEL maintainer="Hiromu Hota <hiromu.hota@hal.hitachi.com>"
+USER root
+
+RUN apt-get update && apt-get install -y \
+    libxml2-dev \
+    libxslt-dev \
+    python-matplotlib \
+    poppler-utils \
+    postgresql-client \
+    libmagickwand-dev \
+    ghostscript \
+ && rm -rf /var/lib/{apt,dpkg,cache,log}/
+RUN rm /etc/ImageMagick-6/policy.xml
+
+RUN pip install torch==0.4.1
+RUN pip install \
+    fonduer==0.3.4 \
+    matplotlib \
+    snorkel-metal==0.2.4
+
+RUN python -m spacy download en
+RUN conda install -y -c conda-forge ipywidgets
+
+USER $NB_UID
+
+COPY --chown=jovyan:users hardware hardware
+COPY --chown=jovyan:users hardware_image hardware_image
+COPY --chown=jovyan:users intro intro
```

Comment:
Having never used Docker, I'm not qualified to do a code review. But, I do notice several places that seem very specific to a particular user like `jovyan` being used here and in `docker-compose.yml`. I'm not sure if this is necessary/intended or if there are more generalizable ways to do this.

Reply:
`jovyan` is originally defined by https://github.com/jupyter/docker-stacks/blob/master/base-notebook/Dockerfile#L10 in the base Docker image.
Note the Docker image inheritance is the current Docker image (say `hiromuhota/fonduer-tutorials`) -> `jupyter/minimal-notebook` -> `jupyter/base-notebook` -> `ubuntu:bionic-20180526`.
In `jupyter/base-notebook`, the jupyter server is ran by `jovyan`. That's why I changed the owner of these folder/files as `jovyan`.

Reply:
Ah, I see. Makes sense.